compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -3335,9 +3359,10 @@ private ParserRuleContext getNextRuleForKeywords(ParserRuleContext currentCtx, i
                 return ParserRuleContext.SEMICOLON;
             case FUNCTION_KEYWORD:
                 parentCtx = getParentContext();
-                if (parentCtx == ParserRuleContext.ANON_FUNC_EXPRESSION ||
-                        parentCtx == ParserRuleContext.FUNC_TYPE_DESC) {
+                if (parentCtx == ParserRuleContext.ANON_FUNC_EXPRESSION) {
                     return ParserRuleContext.OPEN_PARENTHESIS;
+                } else if (parentCtx == ParserRuleContext.FUNC_TYPE_DESC) {
+                    return ParserRuleContext.FUNC_TYPE_FUNC_KEYWORD_RHS;
```

Comment:
We can directly give `FUNC_TYPE_FUNC_KEYWORD_RHS_START` and remove the check inside `getNextRuleForFuncTypeFuncKeywordRhs`

Reply:
updated. thanks!